currency-rate/cmd/main.go

```diff
@@ -0,0 +1,170 @@
+	cronRate "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/cron/rate"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/models"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/rate/fetchers"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/server"
+	serverCfg "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/server/config"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/server/service"
+	transportCfg "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport/config"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/cron"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/database"
+	dbCfg "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/database/config"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/settings"
+)
+
+const (
+	ccFrom = "USD"
+	ccTo   = "UAH"
+)
+
+var appConfig appCfg.Config
+
+const eventTimeout = 5 * time.Second
+
+type UserRepoDecorator struct {
+	*models.UserRepository
+	producer *userProducer.Producer
+}
+
+func (u *UserRepoDecorator) Create(user *models.User) error {
+	if err := u.UserRepository.Create(user); err != nil {
+		return err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), eventTimeout)
+	defer cancel()
+	if err := u.producer.SendSubscribe(ctx, user.Email); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (u *UserRepoDecorator) Delete(user *models.User) error {
+	if err := u.UserRepository.Delete(user); err != nil {
+		return err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), eventTimeout)
+	defer cancel()
+	if err := u.producer.SendUnsubscribe(ctx, user.Email); err != nil {
+		return err
+	}
+	return nil
+}
+
+func InitDatabase() (*database.DB, error) {
+	db, err := database.New(dbCfg.NewFromEnv())
+	if err != nil {
+		return nil, err
+	}
+	if err := db.Migrate(&models.User{}); err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+func InitLogger() *slog.Logger {
+	loggerOptions := &slog.HandlerOptions{}
+	if appConfig.Debug {
+		loggerOptions.Level = slog.LevelDebug
+	}
+	logger := slog.New(
+		slog.NewTextHandler(os.Stdout, loggerOptions),
+	).With(slog.Any("service", "currency-rate"))
+	return logger
+}
+
+func InitFetchers() []service.RateFetcher {
+	nbuFetcher := fetchers.NewNBURateFetcher()
+	currencyBeaconFetcher := fetchers.NewCurrencyBeaconFetcher(
+		appConfig.CurrencyBeaconAPIKey,
+	)
+	return []service.RateFetcher{currencyBeaconFetcher, nbuFetcher}
+}
+
+func InitCron(transportConfig transportCfg.Config, fetcher service.RateFetcher) *cron.Manager {
```

Comment:
if this is a function constructor, maybe name it as `NewCron()` according to the convention?